lib/AST/ASTContext.cpp

```diff
@@ -5166,45 +5187,66 @@ CanGenericSignature ASTContext::getSingleGenericParameterSignature() const {
   return canonicalSig;
 }
 
-// Return the signature for an opened existential. The opened archetype may have
-// a different set of conformances from the corresponding existential. The
-// opened archetype conformances are dictated by the ABI for generic arguments,
-// while the existential value conformances are dictated by their layout (see
-// Type::getExistentialLayout()). In particular, the opened archetype signature
-// does not have requirements for conformances inherited from superclass
-// constraints while existential values do.
-CanGenericSignature ASTContext::getOpenedArchetypeSignature(Type type) {
+Type OpenedArchetypeType::getSelfInterfaceTypeFromContext(const DeclContext *useDC) {
+  auto typeContext = useDC->getInnermostTypeContext();
+  if (typeContext && typeContext->getSelfProtocolDecl()) {
+    return typeContext->getProtocolSelfType();
```

Comment:
I'm not sure this is right. Imagine you're in a protocol extension context where the outer signature is <Self where Self : P>, and you form an existential `any Q`. You'd want the existential signature to be `<Self, Self2 where Self : P, Self2 : Q>`.